tribe-ext-recurring-cleanup/index.php

```diff
@@ -0,0 +1,158 @@
+ */
+defined( 'WPINC' ) or die;
+
+// Do not load directly.
+if ( ! defined( 'ABSPATH' ) ) { die( '-1' ); }
+// Do not load unless Tribe Common is fully loaded.
+if ( ! class_exists( 'Tribe__Extension' ) ) { return; }
+
+/**
+ * Extension main class, class begins loading on init() function.
+ */
+class Tribe__Extension__Recurring_Cleanup extends Tribe__Extension {
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 */
+	public function construct() {
+		$this->add_required_plugin( 'Tribe__Events__Main' );
+		$this->add_required_plugin( 'Tribe__Events__Pro__Main' );
+
+		$this->set_url( 'https://theeventscalendar.com/extensions/the-events-calendar-pro-cleanup-recurring-events/' );
+	}
+
+	/**
+	 * Extension initialization and hooks.
+	 */
+	public function init() {
+		add_action( 'tool_box', array( $this, 'output_cleanup_box' ) );
+	}
+
+	/**
+	 * Echoes the admin UI.
+	 */
+	public function output_cleanup_box() {
+
+		// Process delete event form.
+		if (
+			isset( $_POST['tribe-recurring-cleanup-eventid'] )&&
+			isset( $_POST['tribe-recurring-cleanup-backup-confirmation'] ) &&
+			isset( $_POST['tribe-recurring-cleanup-submit'] ) &&
+			check_ajax_referer( 'tribe-recurring-cleanup' )
+		) {
+			$event_id = intval( $_POST['tribe-recurring-cleanup-eventid'] );
+			$this->delete_recurrence( intval( $_POST['tribe-recurring-cleanup-eventid'] ) );
+			$notifications = __( 'Recurrences deleted for event ID ', 'tribe-extension' ) . $event_id;
+		}
+
+		$recurrence_table_headings = array(
+			__( 'Event', 'tribe-extension' ),
+			__( 'ID', 'tribe-extension' ),
+			__( '# of recurrences', 'tribe-extension' ),
+		);
+		$recurrence_table_cells = $this->get_recurring_event_table_cells();
+		$simple_table = new Tribe__Simple_Table( $recurrence_table_cells, $recurrence_table_headings );
+		$simple_table->table_attributes = array( 'cellspacing' => '0', 'cellpadding' => '5' );
+		$simple_table->html_escape_td_values = false;
+
+		$recurrences_table = $simple_table->output_table();
+
+		require_once dirname( __FILE__ ) . '/src/Tribe/admin-views/recurring-cleanup-module.php';
+	}
+
+	/**
+	 * Returns query containing events with the most recurrences.
+	 *
+	 * Shows how many recurrences each recurring event has.
+	 * Includes events from all statuses including in the trash.
+	 *
+	 * @return array|null|object
+	 */
+	protected function get_recurring_event_list() {
+		global $wpdb;
+
+		$count_recurrences_sql = "
+		SELECT 
+			`post_title` as title,
+			`post_parent` as event_id, 
+			COUNT(*) As recurrences 
+		FROM 
+			{$wpdb->posts}
+		WHERE 
+			`post_parent` <> 0 
+			AND `post_type` = 'tribe_events' 
+		GROUP BY 
+			`post_parent` 
+		ORDER BY 
+			`recurrences` DESC";
+
+		return $wpdb->get_results( $count_recurrences_sql, ARRAY_A );
+	}
+
+	/**
+	 * Gets a list of events with many recurrences.
+	 *
+	 * @return array Each one an HTML <a> link to an event series.
```

Comment:
would recommend using `"link element"` instead of `<a>` in this docblock comment per WordPress coding standards → https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#summary-formerly-short-description